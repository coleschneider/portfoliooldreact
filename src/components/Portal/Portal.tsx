import * as React from 'react'
import { createPortal } from 'react-dom'
import usePortal from '../../hooks/usePortal'

interface Props {
  id: string
  children: React.ReactNode
}
function Portal(props: Props) {
  const target = usePortal(props.id)
  return createPortal(props.children, target)
}

export default Portal
