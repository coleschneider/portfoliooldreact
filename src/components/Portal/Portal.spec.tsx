import React from 'react'
import { render } from '@testing-library/react'
import Portal from './Portal'

const createRootElement = (id: string) => {
  const rootContainer = document.createElement('div')
  rootContainer.setAttribute('id', id)
  document.body.appendChild(rootContainer)
  return rootContainer
}
describe('Portal', () => {
  const id = 'portal'
  let rootContainer: HTMLDivElement
  beforeAll(() => {
    rootContainer = createRootElement('portal')
  })
  it('should match the snapshot', () => {
    const Wrapper = shallow(<Portal id="portal">Test</Portal>)
    expect(Wrapper).toMatchSnapshot()
  })
  it('should show the modals children', () => {
    const { getByText, debug } = render(
      <Portal id="portal">
        <div>Test</div>
      </Portal>,
    )
    expect(getByText('Test')).toBeTruthy()
  })
})
