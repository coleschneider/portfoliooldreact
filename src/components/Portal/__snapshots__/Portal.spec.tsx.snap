// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal should match the snapshot 1`] = `
<Portal
  containerInfo={<div />}
>
  Test
</Portal>
`;
